app.js

```diff
@@ -0,0 +1,149 @@
+}
+
+function getShipRandomPosition() 
+{
+    return { 
+        x : Math.floor(Math.random() * 10),
+        y : Math.floor(Math.random() * 10),
+        isHorizontal : isHorizontal = Math.random() > 0.5 
+    }
+}
+
+function createShipCoordinates(squaresCount)
+{
+    var shipCoordinates = [];
+    
+    for(let ship = 0; ship < squaresCount.length; ship++)
+    {
+        var position = getShipRandomPosition();
+    
+        if (position.isHorizontal)
+        {
+            if(10 - squaresCount[ship] >= position.x)
+            {
+                for (let length = 0; length < squaresCount[ship]; length++)
+                {   
+                    shipCoordinates.push("ship-part-" + position.y + "-" + position.x);
+                    position.x++;
+                }
+            }
+            else
+            {
+                position.x = 10 - squaresCount[ship];
+                for (let length = 0; length < squaresCount[ship]; length++)
+                {   
+                    shipCoordinates.push("ship-part-" + position.y + "-" + position.x);
+                    position.x++;
+                }
+            }
+        }
+        else
+        {
+            if(10 - squaresCount[ship] >= position.y)
+            {
+                for (let length = 0; length < squaresCount[ship]; length++)
+                {   
+                    shipCoordinates.push("ship-part-" + position.y + "-" + position.x);
+                    position.y++;
+                }
+            }
+            else
+            {
+                position.y = 10 - squaresCount[ship];
+                for (let length = 0; length < squaresCount[ship]; length++)
+                {   
+                    shipCoordinates.push("ship-part-" + position.y + "-" + position.x);
+                    position.y++;
+                }
+            }
+        }
+    }
+
+    return shipCoordinates;
+}
+
+function createShipSquares(shipCoordinates, battleField)
+{
+    for (let i = 0; i < shipCoordinates.length; i++) 
+    {
+        for (let x = 0; x < 10; x++)
+        {
+            for(let y = 0; y < 10; y++)
+            {
+                if (battleField == getContainer("player-grid-container"))
+                {
+                    const square = document.getElementById("user square-" + x + "-" + y);
+                    if("ship-part-" + x + "-" + y === shipCoordinates[i])
+                    {
+                        square.className += " square-ship";
+                    }
+                }
+                else if (battleField == getContainer("computer-grid-container"))
+                {
+                    const square = document.getElementById("computer square-" + x + "-" + y);
+                    if("ship-part-" + x + "-" + y === shipCoordinates[i])
+                    {
+                        square.className += " square-ship";
+                    }
+                }
+            }
+        }
+    }
+}
+
+function createUserShips()
```

Comment:
createUserShips and createComputerShips methods look 90% identical
create separate method for the similar part